include/onnxruntime/core/framework/data_types_internal.h

```diff
@@ -558,10 +559,11 @@ class ContainerChecker {
         ORT_ENFORCE(++index < c.size(), "Sequence is missing type entry for its element");
         constexpr int32_t prim_type = ToTensorProtoElementType<T>();
         // Check if this is a primitive type and it matches
-        ORT_IF_CONSTEXPR (prim_type != ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED) {
+        ORT_IF_CONSTEXPR(prim_type != ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED) {
```

Comment:
>ORT_IF_CONSTEXPR(prim_type  [](http://example.com/codeflow?start=8&length=27)

Is Optional going to be contained in vectors or maps? I thought, you explicitely do not support them.

Reply:
It is not supported per the spec